### DISCUSSION TITLE
Mapset remix

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#23613] [Ideas]
Hello. There are many cases when the mappers map many versions of the same song. I have an idea of what is called a "mapset remix".
For example, we have ranked Dragonforce song with 1 diff that is 8*. But I, as a player, can't play 8* maps. Imagine that there are were a button (on the map page) where I can see all diffs mapped on this song (mp3). They can have worse quality (bcs isn't ranked) but at least I can play them. So in this case I only have to download small file without mp3/storyboard/hitsounds. So, as player, it will be very good quality of life change.
As a mapper it will be good as well - don't need to make a map from scratch / dig through files of the original map to make your own. Also, you can make the upload limit on remix diffs much higher, because they don't consume much space. Also, you can easily compare your diff with other people's similar diffs on this song.
As a server holder, it will be good space-wise. Imagine if all 100 versions of harumachi clover were bound to one mp3/background/hitsounds. I'm pretty sure that space requirements would be like 2x lower. 
Ye, there are already many mapsets of the same song. Ideally, we want to make them remixes, not full mapsets, but it gonna be pretty hard so I don't know about it.
Peppy and others, what do you think of this? (ofc it's not a matter of first necessity)

**Comment by bdach** [comment]
If the primary rationale for this feature is saving disk space, then you might want to know that the game already internally deduplicates files because of how lazer's file storage is constructed. If you import two maps with the exact same .mp3, only one copy of it will be stored on disk. https://github.com/ppy/osu/wiki/User-file-storage explains the details.

**Reply by Givikap120** [reply]
I mean not only user disk space but server disk space too

**Comment by Rekunan** [comment]
> They can have worse quality (bcs isn't ranked)

I think it's better they were only leaderboarded (ranked & loved) sets.
